src/Advection/adapt_advection_order.jl

```diff
@@ -0,0 +1,111 @@
+using Oceananigans.Grids: topology
+
+"""
+    adapt_advection_order(advection, grid::AbstractGrid)
+
+Adapts the advection operator `advection` based on the grid `grid` by adjusting the order of advection in each direction.
+For example, if the grid has only one point in the x-direction, the advection operator in the x-direction is set to first order
+upwind or 2nd order centered scheme, depending on the original user-specified advection scheme. A high order advection sheme 
+is reduced to a lower order advection scheme if the grid has fewer points in that direction.
+
+# Arguments
+- `advection`: The original advection scheme.
+- `grid::AbstractGrid`: The grid on which the advection scheme is applied.
+
+If the order of advection is changed in at least one direction, the adapted advection scheme with adjusted advection order returned 
+by this function is a `FluxFormAdvection`.
+"""
+function adapt_advection_order(advection, grid::AbstractGrid)
+    advection_x = adapt_advection_order(advection, size(grid, 1), grid)
+    advection_y = adapt_advection_order(advection, size(grid, 2), grid)
+    advection_z = adapt_advection_order(advection, size(grid, 3), grid)
+
+    # Check that we indeed changed the advection operator
+    changed_x = advection_x != advection
+    changed_y = advection_y != advection
+    changed_z = advection_z != advection
+
+    new_advection = FluxFormAdvection(advection_x, advection_y, advection_z)
+    changed_advection = any((changed_x, changed_y, changed_z))
+
+    if changed_x
+        @info "Using the advection scheme $(summary(new_advection.x)) in the x-direction because size(grid, 1) = $(size(grid, 1))"
+    end
+    if changed_y
+        @info "Using the advection scheme $(summary(new_advection.y)) in the y-direction because size(grid, 2) = $(size(grid, 2))"
+    end
+    if changed_z
+        @info "Using the advection scheme $(summary(new_advection.z)) in the x-direction because size(grid, 3) = $(size(grid, 3))"
+    end
+
+    return ifelse(changed_advection, new_advection, advection)
+end
+
+function adapt_advection_order(advection::FluxFormAdvection, grid::AbstractGrid)
+    advection_x = adapt_advection_order(advection.x, size(grid, 1), grid)
+    advection_y = adapt_advection_order(advection.y, size(grid, 2), grid)
+    advection_z = adapt_advection_order(advection.z, size(grid, 3), grid)
```

Comment:
```suggestion
    original_x_advection = x_advection(advection)
    original_y_advection = y_advection(advection)
    original_z_advection = z_advection(advection)

    adapted_x_advection = adapt_advection_order(original_advection, size(grid, 1), grid)
    adapted_y_advection = adapt_advection_order(original_advection, size(grid, 2), grid)
    adapted_z_advection = adapt_advection_order(original_advection, size(grid, 3), grid)
```

You just need

```julia
x_advection(flux_form::FluxFormAdvection) = flux_form.x
x_advection(advection) = advection
```

and then you can avoid all the copy-pasting